### ISSUE TITLE
azurerm: Import requires Subscription ID, Client ID and Tenant ID set as environment variables vs configuration

### ISSUE THREAD

**Issue by pearcec** [hashicorp/terraform#8191]
### Terraform Version

$ terraform --version
Terraform v0.7.0
### Affected Resource(s)
- azurerm_resource_group
### Terraform Configuration Files

``` hcl
provider "azurerm" {
  tenant_id = "00000000-0000-0000-0000-000000000000"
  subscription_id = "00000000-0000-0000-0000-000000000000"
  client_id = "00000000-0000-0000-0000-000000000000"
}
```
### Expected Behavior

What should have happened?

$ terraform  import azurerm_resource_group.testrg /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg
azurerm_resource_group.testrg: Import complete!
Import success! The resources imported are shown above. These are
now in your Terraform state. Import does not currently generate
configuration, so you must do this next. If you do not create configuration
for the above resources, then the next `terraform plan` will mark
them for destruction.
### Actual Behavior

What actually happened?

$ terraform  import azurerm_resource_group.testrg /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg
Error importing: 1 error(s) occurred:
- 3 error(s) occurred:
- Subscription ID must be configured for the AzureRM provider
- Client ID must be configured for the AzureRM provider
- Tenant ID must be configured for the AzureRM provider
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. Run the above configuration without export the required variables

export ARM_CLIENT_ID=""
export ARM_SUBSCRIPTION_ID=""
export ARM_TENANT_ID=""

**Comment by stack72** [comment]
Hi @pearcec 

Thanks for opening the issue here. sorry this is causing some pain for you. Unfortunately, this is true right now. I am going to label this as an enhancement to the core and add some documentation for this for now to note that this is the case

We will endeavor to get this taken care of soon

Paul

**Comment by mitchellh** [comment]
Fixed by #9809 (not yet merged but will be shortly)
